### PR TITLE
feat(blocks): add plain text block adapter for linked and synced doc block

### PR DESCRIPTION
**Pull request by donteatfriedrice**
[BS-2040](https://linear.app/affine-design/issue/BS-2040/补齐-linked-doc-和-synced-doc-plain-text-adapter)

### CODE REVIEW

**Comment by donteatfriedrice** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/8848?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#8848** <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/8848?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/8848?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#8845** <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/8845?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: packages/affine/block-embed/src/embed-linked-doc-block/adapters/plain-text.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    fromBlockSnapshot: {
+      enter: (o, context) => {
+        const { configs, textBuffer } = context;
+        // Parse as link
+        if (!o.node.props.pageId) {
+          return;
+        }
+        const title = configs.get('title:' + o.node.props.pageId) ?? 'untitled';
+        const { mode, blockIds, elementIds } = o.node.props;
+        const baseUrl = configs.get('docLinkBaseUrl') ?? '';
```
Comment (fundon) [inline packages/affine/block-embed/src/embed-linked-doc-block/adapters/plain-text.ts:23]: use `GenerateDocUrlProvider`, see https://github.com/toeverything/blocksuite/pulls

File: packages/affine/block-embed/src/embed-linked-doc-block/adapters/plain-text.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        const search = toURLSearchParams({
+          mode: mode as string,
+          blockIds: blockIds as string[],
+          elementIds: elementIds as string[],
+        });
```
Comment (fundon) [inline packages/affine/block-embed/src/embed-linked-doc-block/adapters/plain-text.ts:28]: https://github.com/toeverything/blocksuite/blob/73e85f63a06e869021747854fe4bc23bd8310064/packages/affine/model/src/consts/doc.ts#L7-L14

Missing some fields

File: packages/affine/block-embed/src/embed-linked-doc-block/adapters/plain-text.ts (hunk 1 above)
Comment (donteatfriedrice) [reply packages/affine/block-embed/src/embed-linked-doc-block/adapters/plain-text.ts]: These adapters may be used without BlockStd and providers.

**Review by L-Sun — APPROVED** [review]

**Pull request merged**